File: front/src/types/Skill.ts
export interface Skill {
    id: number;
    name: string;
    category?: string;
    description?: string;
}

export interface UserSkill {
    id: number;
    userId: number;
    skillId: number;
    skill: Skill;
    type: 'OFFER' | 'REQUEST';
    level?: number;  // 1-5
}

export interface AddUserSkillRequest {
    skillId: number;
    type: 'OFFER' | 'REQUEST';
    level?: number;
}

File: back/src/main/java/org/example/skillandyou/entity/Skill.java
package org.example.skillandyou.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import jakarta.persistence.ManyToMany;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;


@Entity
@Table(name ="skills")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Skill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Le nom de la compétence est obligatoire")
    @Column(nullable = false, unique = true, length = 100)
    private String name;

    @NotBlank(message = "La catégorie est obligatoire")
    @Column(nullable = false, length = 50)
    private String category;

    @Column(columnDefinition = "TEXT")
    private String description;

    @OneToMany(mappedBy = "skill", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<UserSkill> userSkills = new ArrayList<>();


}
